context/metadata.ts

```diff
@@ -26,11 +32,37 @@ async function fetchVersion(
   }
 ): Promise<version> {
   try {
-    let metar = await metadata.tzip16().getMetadata();
-    let version = metar.metadata.version!;
-    return dispatch[version] || "unknown version";
+    const version = await metadata
+      .tzip16()
+      .getMetadata()
+      .then(metadata => metadata.metadata.version ?? "unknown version")
+      .catch(_ =>
+        fetch(`${API_URL}/v1/contracts?address=${metadata.address}`)
+          .then(r => r.json())
+          .then(
+            ([{ typeHash, codeHash }]: {
+              typeHash: number;
+              codeHash: number;
+            }[]) => VERSION_HASH[`${typeHash}${codeHash}`] ?? "unknown version"
```

Comment:
would it be better like? that would be easy to understand.
```suggestion
            }[]) => VERSION_HASH[[${typeHash},${codeHash}]] ?? "unknown version"
```